### PR TITLE
Preemptive Multitasking Additions

### PR DESCRIPTION
**Pull request by BitterB0NG0**

### CODE REVIEW

File: src/os/kernel/kernel.c [hunk 1]
```diff
@@ -87,6 +87,12 @@ uint kernel_create_process(void (*pointer_to_task_function)(void), int necessity
 	new_node->status = READY;
 	new_node->next = NULL;
 
+	tack_words += new_node.stack_size - 17;
```
Comment (BjornTheProgrammer) [inline src/os/kernel/kernel.c]: task_words doesn't seem to be defined anywhere else. I assume it is supposed to be stack_words, additionally, it has never been defined yet.

File: src/os/kernel/systick.s [hunk 2]
```diff
@@ -0,0 +1,21 @@
+.text
```
Comment (BjornTheProgrammer) [inline src/os/kernel/systick.s:1]: This file needs to be included in cmake and called within kernel.c

File: src/os/kernel/kernel.c [hunk 3]
```diff
@@ -7,6 +7,11 @@ i.e. prevent crashing of entire system from process accessing wrong memory or nu
 #include "malloc.h"
 #include <string.h>
 
+
```
Comment (BjornMediaPro) [inline src/os/kernel/kernel.c:10]: Need to define what __initialize_context_switch does

**Review by BjornMediaPro — CHANGES_REQUESTED** [review]
The following needs to be updated

**Review by BjornTheProgrammer — CHANGES_REQUESTED** [review]

File: src/os/kernel/kernel.c [hunk 4]
```diff
@@ -7,6 +7,11 @@ i.e. prevent crashing of entire system from process accessing wrong memory or nu
...
+unsigned int *__initialize_context_switch(unsigned int *stack) {
```
Comment (BjornTheProgrammer) [inline src/os/kernel/kernel.c:11]: Need to define what __initialize_context_switch does

**Review by BjornTheProgrammer — CHANGES_REQUESTED** [review]

**Pull request closed without merging**